packages/hoppscotch-backend/src/user-history/user-history-feature-flag.guard.ts

```diff
@@ -0,0 +1,20 @@
+import { CanActivate, ExecutionContext, Injectable } from '@nestjs/common';
+import { USER_HISTORY_FEATURE_FLAG_DISABLED } from 'src/errors';
+import { InfraConfigService } from 'src/infra-config/infra-config.service';
+import { throwErr } from 'src/utils';
+import * as E from 'fp-ts/Either';
+
+@Injectable()
+export class UserHistoryFeatureFlagGuard implements CanActivate {
```

Comment:
is a guard required for this specific use case? Can't we just handle it in sthe ervice method?

Reply:
Since we are enabling or disabling entire features (e.g., `createHistory`, `starringHistory`, `deleteHistory`, `getHistory`, `subscription`, etc.), implementing a **guard** can simplify code maintenance and cleaner. This approach avoids the repetition of `if` conditions in every service function, centralizing the logic for feature toggling.